citizen.ts

```diff
@@ -344,3 +391,125 @@ export function scheduleRendezvous(
 
   return `${targetContractSource}\n\n${context}\n\n${tests}`;
 }
+
+/**
+ * Utility function that restores the test wallets' initial sBTC balances in
+ * the re-initialized first-class citizenship simnet.
+ *
+ * @param simnet The simnet instance.
+ * @param sbtcBalancesMap A map containing the test wallets' balances to be
+ * restored.
+ */
+const restoreSbtcBalances = (
+  simnet: Simnet,
+  sbtcBalancesMap: Map<string, number>
+) => {
+  // To deposit sBTC, one needs a txId and a sweep txId. A deposit transaction
+  // must have a unique txId and sweep txId. A set will be used to track the
+  // already used ids. This set will be used in `getUniqueRandomHex`.
+  const usedTxIds: Set<string> = new Set();
+
+  // For each address present in the balances map, restore the balance.
+  Array.from(sbtcBalancesMap.entries())
+    // Re-assure the map does not contain nil balances.
+    .filter(([_, balance]) => balance !== 0)
+    .forEach(([address, balance]) => {
+      const txId = getUniqueRandomHex(usedTxIds);
+      const sweepTxId = getUniqueRandomHex(usedTxIds);
+      mintSbtc(simnet, balance, address, txId, sweepTxId);
+    });
+};
+
+/**
+ * Utility function to deposit an amount of sBTC to a Stacks address.
+ *
+ * @param simnet The simnet instance.
+ * @param amountSats The amount to mint in sats.
+ * @param recipient The Stacks address to mint sBTC to.
+ * @param txId A unique hex to use for the deposit.
+ * @param sweepTxId A unique hex to use for the deposit.
+ */
+const mintSbtc = (
+  simnet: Simnet,
+  amountSats: number,
+  recipient: string,
+  txId: string,
+  sweepTxId: string
+) => {
+  const blockHeight = simnet.blockHeight;
+
+  // Calling `get-burn-header` only works for past block heights. We mine one
+  // empty Bitcoin block along with a Stacks one to make sure this happens.
+  simnet.mineEmptyBurnBlocks(1);
+
+  const burnHash = simnet.callReadOnlyFn(
+    "SM3VDXK3WZZSA84XXFKAFAF15NNZX32CTSG82JFQ4.sbtc-deposit",
+    "get-burn-header",
+    [
+      // (height uint)
+      uintCV(blockHeight),
+    ],
+    simnet.deployer
+  ).result as OptionalCV<BufferCV>;
+
+  if (burnHash === null || burnHash.type === ClarityType.OptionalNone) {
+    throw new Error("Something went wrong trying to retrieve the burn header.");
+  }
+
+  const completeDepositTx = cvToJSON(
+    simnet.callPublicFn(
+      "SM3VDXK3WZZSA84XXFKAFAF15NNZX32CTSG82JFQ4.sbtc-deposit",
+      "complete-deposit-wrapper",
+      [
+        // (txid (buff 32))
+        bufferCV(hexToBytes(txId)),
+        // (vout-index uint)
+        uintCV(1),
+        // (amount uint)
+        uintCV(amountSats),
+        // (recipient principal)
+        principalCV(recipient),
+        // (burn-hash (buff 32))
+        burnHash.value,
+        // (burn-height uint)
+        uintCV(blockHeight),
+        // (sweep-txid (buff 32))
+        bufferCV(hexToBytes(sweepTxId)),
+      ],
```

Comment:
We tend to prefer Cl helpers; `Cl.uint`, `Cl.buffer` etc (can also avoid the hexToBytes`.
But that's just an alternative, whatever you prefer is fine